products/sentinel_one.py

```diff
@@ -153,15 +99,136 @@ def _authenticate(self):
 
-        # test API key by retrieving the sensor count, which is a fast operation
-        data = self._session.get(self._build_url('/web/api/v2.1/agents/count'),
-                                 headers=self._get_default_header(),
-                                 params=self._get_default_body()).json()
-        if 'errors' in data:
-            if data['errors'][0]['code'] == 4010010:
-                raise ValueError(f'Failed to authenticate to SentinelOne: {data}')
-            else:
-                raise ValueError(f'Error when authenticating to SentinelOne: {data}')
+        # generate a list of site_ids based on config file and cmdline input
+        # this will also test API keys as it goes
+        self._get_site_ids(self.site_id, self.account_id, self.account_name)
+
+        if len(self._site_ids) < 1 and len(self._account_ids) < 1:
+            raise ValueError(f'S1 configuration invalid, specify a site_id, account_id, or account_name')
+
+    def _get_site_ids(self, site_id, account_id, account_name):
+        config = configparser.ConfigParser()
+        config.read(self.creds_file)
+
+        # check if any cmdline stuff was input - that will take precedence over config file stuff
+        site_ids = (site_id) if site_id else list()
+        account_ids = (account_id) if account_id else list()
+        account_names = (account_name) if account_name else list()
+
+        if not site_ids and not account_ids and not account_names:
+            # extract account/site ID from configuration if set
+            if 'account_id' in config[self.profile]:
+                for id in config[self.profile]['account_id'].split(','):
+                    if id not in account_ids:
+                        account_ids.append(id.strip())
+
+            if 'site_id' in config[self.profile]:
+                for id in config[self.profile]['site_id'].split(','):
+                    if id not in site_ids:
+                        site_ids.append(id.strip())
+
+            if 'account_name' in config[self.profile]:
+                for name in config[self.profile]['account_name'].split(','):
+                    if name not in account_names:
+                        account_names.append(name.strip())
+
+        # determine site and account IDs to query (default is all)
+        self._site_ids = list()
+        self._account_ids = list()
+
+        if account_ids: # verify provided account IDs are valid
+            # create batch of 10 account IDs per call
+            counter = 0
+            temp_list = []
+            i = 0
+            while i < len(account_ids):
+                temp_list.append(account_ids[i])
+                counter += 1
+                if counter == 10 or i == len(account_ids) - 1:
+                    response = self._get_all_paginated_data(self._build_url(f'/web/api/v2.1/accounts'),
+                                                            params={'ids': ','.join(temp_list)},
```

Comment:
```suggestion
                                                            params={'states': "active", 'ids': ','.join(temp_list)},
```
only return active sites. Any expired/deleted sites will be logged in mismatch check.